packages/adapters-library/src/adapters/pendle/products/yield-token/pendleYieldTokenAdapter.ts

```diff
@@ -103,14 +130,26 @@ export class PendleYieldTokenAdapter implements IProtocolAdapter {
         ...yt,
         underlyingTokens: [sy],
         marketAddress,
+        expiry: value.expiry,
       }
     })
   }
 
   async getPositions(input: GetPositionsInput): Promise<ProtocolPosition[]> {
+    const tokens = await filterMapAsync(
+      await this.getProtocolTokens(),
+      async (protocolToken) => {
+        if (await this.isExpiredAtBlock(protocolToken.expiry)) {
+          return undefined
+        }
+
+        return protocolToken
+      },
+    )
```

Comment:
At least add a TODO here, because when we can update the metadata db dynamically we 1000000% don't want to do this, as it's just delaying fetching positions whilst waiting for block info.

Reply:
Hey,

I see - you are saying we run metadata daily then save isExpired to true in db, interesting 

You found a good use case to run the metadata daily.